clang/lib/Sema/SemaTemplateVariadic.cpp

```diff
@@ -281,7 +283,115 @@ namespace {
       return inherited::TraverseLambdaCapture(Lambda, C, Init);
     }
   };
-}
+
+
+  class CollectExpandedParameterPacksVisitor
```

Comment:
Rather than catching them like this... could we save the packs when parsing/instantiating?  It would be preferential to store the parameter packs (perhaps as a potential list?) in the Lambda scope info objects as we parse, then again as we instantiate.

Reply:
As a general concern, I think various parts of our pack handling would be cleaner if the parser tracked the set of packs used in a pack expandable region and annotated them on the pack expansion construct, rather than recursively walking the AST to reconstruct the list after the fact. (The latter is working, but it's fragile and we've had various bugs with the walk not quite finding everything for various reasons. It's also inefficient to walk the complete expanded region to search for the packs -- but given that we'll need to do that walk anyway when we actually expand, it's not asymptotically bad, except in the case where the packs are empty.)